kornia/onnx/sequential.py

```diff
@@ -0,0 +1,150 @@
+from typing import Optional, Union
+
+from kornia.core.external import numpy as np
+from kornia.core.external import onnx
+from kornia.core.external import onnxruntime as ort
+
+__all__ = ["ONNXSequential"]
+
+
+class ONNXSequential:
+    """ONNXSequential to chain multiple ONNX operators together.
+
+    Args:
+        *args:
+            A variable number of ONNX models (either ONNX ModelProto objects or file paths).
+        providers:
+            A list of execution providers for ONNXRuntime (e.g., ['CUDAExecutionProvider', 'CPUExecutionProvider']).
+        session_options:
+            Optional ONNXRuntime session options for optimizing the session.
+        io_maps:
+            An optional list of list of tuples specifying input-output mappings for combining models.
+            If None, we assume the default input name and output name are "input" and "output" accordingly, and
+            only one input and output node for each graph.
+            If not None, `io_maps[0]` shall represent the `io_map` for combining the first and second ONNX models.
+    """
+
+    def __init__(
+        self,
+        *args: Union[onnx.ModelProto, str],  # type:ignore
```

Comment:
Would someone useful to support in-memory models too, to simplify to go from a raw model + pre/post processing to the combined pipeline

Reply:
isn't the loaded onnx model in-memory? if not, how to support?